swap_meet/vendor.py

```diff
@@ -1,2 +1,56 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not (my_item in self.inventory and their_item in other_vendor.inventory):
+            return False
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if not (self.inventory and other_vendor.inventory):
+            return False
+        return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
Nice reuse of a function you already wrote! However, notice that due to the remove calls in swap_items, this will be O(n) where n is the lengths of the inventories, since we'll have to shift the whole remaining part of the inventories forward. 

If you write your own implementation here, you could do this swap in constant time.

There is a way to swap elements in a list using multivariable assignment in python. That would look like: 
```python
a_list = ["cat", "dog"]
a_list[0], a_list[1] = a_list[1], a_list[0]
```

So for this method it would be like: 
```python
self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0] 
```

Why do you think that swapping this way has less time complexity than using the remove() method? Would be good for you to investigate!